File: src/main/java/com/epicdragonworld/gameserver/data/SkillData.java
package com.epicdragonworld.gameserver.data;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.util.HashMap;
import java.util.Map;
import java.util.logging.Logger;

import com.epicdragonworld.gameserver.enums.SkillType;
import com.epicdragonworld.gameserver.managers.DatabaseManager;
import com.epicdragonworld.gameserver.model.holders.SkillHolder;

/**
 * @author Pantelis Andrianakis
 */
public class SkillData
{
	private static final Logger LOGGER = Logger.getLogger(SkillData.class.getName());
	
	private static final String RESTORE_SKILLS = "SELECT * FROM skills";
	
	private static final Map<Long, SkillHolder> _skills = new HashMap<>();
	
	public static void init()
	{
		_skills.clear();
		
		try (Connection con = DatabaseManager.getConnection();
			PreparedStatement ps = con.prepareStatement(RESTORE_SKILLS))
		{
			try (ResultSet rset = ps.executeQuery())
			{
				while (rset.next())
				{
					final int skillId = rset.getInt("skill_id");
					final int skillLevel = rset.getInt("level");
					_skills.put(getSkillHashCode(skillId, skillLevel), new SkillHolder(skillId, skillLevel, Enum.valueOf(SkillType.class, rset.getString("type")), rset.getInt("reuse"), rset.getInt("range"), rset.getInt("param_1"), rset.getInt("param_2")));
				}
			}
		}
		catch (Exception e)
		{
			LOGGER.warning(e.getMessage());
		}
		
		LOGGER.info("SkillData: Loaded " + _skills.size() + " skills.");
	}
	
	private static long getSkillHashCode(int skillId, int skillLevel)
	{
		return (skillId * 100000) + skillLevel;
	}
	
	public static SkillHolder getSkillHolder(int skillId, int skillLevel)
	{
		return _skills.get(getSkillHashCode(skillId, skillLevel));
	}
}

File: EpicDragonWorldServer/Data/SkillData.cs
using MySql.Data.MySqlClient;
using System;
using System.Collections.Generic;

/**
 * Author: Pantelis Andrianakis
 * Date: May 5th 2019
 */
public class SkillData
{
    private static readonly string RESTORE_SKILLS = "SELECT * FROM skills";
    private static readonly Dictionary<long, SkillHolder> SKILLS = new Dictionary<long, SkillHolder>();

    public static void Load()
    {
        Util.PrintSection("Skills");

        SKILLS.Clear();
        try
        {
            MySqlConnection con = DatabaseManager.GetConnection();
            MySqlCommand cmd = new MySqlCommand(RESTORE_SKILLS, con);
            MySqlDataReader reader = cmd.ExecuteReader();
            while (reader.Read())
            {
                int skillId = reader.GetInt32("skill_id");
                int skillLevel = reader.GetInt32("level");
                SKILLS.Add(GetSkillHashCode(skillId, skillLevel), new SkillHolder(skillId, skillLevel, (SkillType)Enum.Parse(typeof(SkillType), reader.GetString("type")), reader.GetInt32("reuse"), reader.GetInt32("range"), reader.GetInt32("param_1"), reader.GetInt32("param_2")));
            }
            con.Close();
        }
        catch (Exception e)
        {
            LogManager.Log(e.ToString());
        }

        LogManager.Log("SkillData: Loaded " + SKILLS.Count + " skills.");
    }

    private static long GetSkillHashCode(int skillId, int skillLevel)
    {
        return (skillId * 100000) + skillLevel;
    }

    public static SkillHolder GetSkillHolder(int skillId, int skillLevel)
    {
        if (!SKILLS.ContainsKey(skillId))
        {
            return null;
        }
        return SKILLS[GetSkillHashCode(skillId, skillLevel)];
    }
}
